### PR TITLE
[Model] Update GAT model code

### PR DESCRIPTION
**Pull request by ylfdq1118**
## Description
- Edge softmax using DGL kernels
- Better implementation to avoid transpose
- Update GAT model code to use latest edge softmax implementation. 
- Implement GAT model for mxnet
- Update accuracy and training speed in readme

## Checklist
- [x] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by yzh119** [comment]
Great, you could also add a row for GAT in the README.

**Comment by jermainewang** [comment]
@ylfdq1118 could you also change mxnet's example?

**Comment by ylfdq1118** [comment]
@yzh119 Done.

@jermainewang mxnet/gat code seems completely out-dated. When you say change mxnet's example, I believe you mean completely rewriting it following pytorch implementation, right?

**Comment by jermainewang** [comment]
> @jermainewang mxnet/gat code seems completely out-dated. When you say change mxnet's example, I believe you mean completely rewriting it following pytorch implementation, right?

Correct.

**Comment by ylfdq1118** [comment]
Done. @jermainewang
